source/tutorials/dev-environment.md

```diff
@@ -28,67 +13,104 @@ app = Flask(__name__)
 
 @app.route("/")
 def hello():
-    return "Hello, Nix!"
+    return {
+        "message": "Hello, Nix!"
+    }
 
 def run():
-    app.run(host="0.0.0.0")
+    app.run(host="0.0.0.0", port=5000)
 
 if __name__ == "__main__":
     run()
 ```
 
-And a `setup.py` script:
+This is a simple Flask application which serves a JSON document with the message
+"Hello, Nix!".
 
-```{code-block} python setup.py
-from setuptools import setup
+For the development environment, create a new file `shell.nix` which creates a
+shell environment to be used for development: 
 
-setup(
-    name='myapp',
-    version='0.1',
-    py_modules=['myapp'],
-    entry_points={
-        'console_scripts': ['myapp = myapp:run']
-    },
-)
-```
+```{code-block} nix shell.nix
+{ pkgs ? import <nixpkgs> {} }:
```

Comment:
This should use the Nixpkgs commit hash from the first pinning tutorial one encounters, otherwise the reader would have to download everything again.